packages/functions/src/context.ts

```diff
@@ -47,11 +48,16 @@ export class ContextProvider {
   private auth: FirebaseAuthInternal | null = null;
   private messaging: MessagingInternal | null = null;
   private appCheck: FirebaseAppCheckInternal | null = null;
+  private serverAppAppCheckToken: string | null = null;
   constructor(
+    readonly app: FirebaseApp,
     authProvider: Provider<FirebaseAuthInternalName>,
     messagingProvider: Provider<MessagingInternalComponentName>,
     appCheckProvider: Provider<AppCheckInternalComponentName>
   ) {
+    if (_isFirebaseServerApp(app) && app.settings.appCheckToken) {
+      this.serverAppAppCheckToken = app.settings.appCheckToken;
+    }
```

Comment:
Similiar to the other SDKs (above) during the initialization of the `AppCheckTokenProvider` that functions implements, check to see if the provided instance of `FirebaseApp` is one of `FirebaseServerApp`, and if it contains an App Check token, then store it locally so it can be returned by the `getToken` method (below).